java/test/apps/startup/StartupActionFactoryScaffold.java

```diff
@@ -0,0 +1,39 @@
+package apps.startup;
+
+import java.util.Locale;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ * The purpose of this class is to test that a class that extends
+ * apps.startup.AbstractStartupActionFactory can use the service
+ * provider apps.startup.StartupActionFactory.
+ * 
+ * This class will be removed when apps.startup.StartupActionFactory
+ * and apps.startup.AbstractStartupActionFactory is removed.
+ * 
+ * Important:
+ * When this class is removed, the string "String" in
+ * apps.startup.Bundle.properties should be removed. That string is needed
+ * for the AbstractActionModelTest test.
+ * 
+ * @author Daniel Bergqvist Copyright (C) 2020
+ */
+@SuppressWarnings("deprecation")
+@ServiceProvider(service = StartupActionFactory.class)
+public class StartupActionFactoryScaffold extends AbstractStartupActionFactory {
+
+    @Override
+    public String getTitle(Class<?> clazz, Locale locale) throws IllegalArgumentException {
+        if (!clazz.equals(String.class)) {
+            throw new IllegalArgumentException();
+        }
+        return Bundle.getMessage(locale, "String"); // NOI18N
```

Comment:
You can write your test without polluting the JMRI sources to do; please return an empty or untranslated String.

Reply:
Fixed. Thanks.